src/Context/ProductListContext.js

```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+import { useContext, createContext, useState, useEffect } from "react";
+const ProductListContext = createContext([]);
+
+const ProductListProvider = ({ children }) => {
+  const [productList, setProductList] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        const {
+          data: { products },
+        } = await axios.get("/api/products");
+        setProductList(products);
+      } catch {
```

Comment:
You can store the error also in a state of there's any. In this way error can be shown in the UI